### PR TITLE
Refactor gateway ClientStreamConsumer

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

Remove dependency to ActorFuture. Instead use CompletableFuture. This is done so that the consumers of this API do not need to know about Actors or actor scheduler. 

We assume the implementation of `ClientStreamConsumer#push` is asynchronous. But we don't enforce it. This should be taken care of in https://github.com/camunda/zeebe/issues/12389

### CODE REVIEW

File: transport/src/main/java/io/camunda/zeebe/transport/stream/impl/ClientStream.java [hunk 1]
```diff
@@ -17,4 +19,22 @@
     AggregatedClientStream<M> serverStream,
     DirectBuffer streamType,
     M metadata,
-    ClientStreamConsumer clientStreamConsumer) {}
+    ClientStreamConsumer clientStreamConsumer) {
+
+  ActorFuture<Void> push(final DirectBuffer payload, final ConcurrencyControl executor) {
+    final ActorFuture<Void> result = executor.createFuture();
```
Comment (deepthidevaki) [inline transport/src/main/java/io/camunda/zeebe/transport/stream/impl/ClientStream.java:25]: Passing the executor here is a bit wierd. But it is done to help use TestActorFuture in tests.

File: transport/src/main/java/io/camunda/zeebe/transport/stream/impl/ClientStream.java [hunk 2]
```diff
@@ -17,4 +19,22 @@
...
+    try {
+      clientStreamConsumer
+          .push(payload)
+          .thenAccept(ignore -> result.complete(null))
+          .exceptionally(
```
Comment (npepinpe) [inline transport/src/main/java/io/camunda/zeebe/transport/stream/impl/ClientStream.java:30]: :wrench: `exceptionally` is used to convert the result of an exception into a success. This isn't really what we want to do here, so I'd rather just use `whenComplete`, e.g.

```java
clientStreamConsumer.push(payload).whenComplete(result);
```

I think `ActorFuture` is a `BiConsumer`, so it should just work.

**Review by npepinpe — APPROVED** [review]
:+1: just one suggestion

**Comment by deepthidevaki** [comment]
bors merge

**Pull request merged**